cardano-api/src/Cardano/Api/Script.hs

```diff
@@ -1170,62 +1027,56 @@ fromShelleyBasedScript  :: ShelleyBasedEra era
-      fromAllegraTimelock TimeLocksInSimpleScriptV2 script
+      ScriptInEra SimpleScriptInAllegra
+        . SimpleScript $ fromAllegraTimelock script
     ShelleyBasedEraMary ->
-      ScriptInEra SimpleScriptV2InMary $
-      SimpleScript SimpleScriptV2 $
-      fromAllegraTimelock TimeLocksInSimpleScriptV2 script
+      ScriptInEra SimpleScriptInMary
+        . SimpleScript $ fromAllegraTimelock script
     ShelleyBasedEraAlonzo ->
       case script of
         Alonzo.TimelockScript s ->
-          ScriptInEra SimpleScriptV2InAlonzo $
-          SimpleScript SimpleScriptV2 $
-          fromAllegraTimelock TimeLocksInSimpleScriptV2 s
+          ScriptInEra SimpleScriptInAlonzo
+            . SimpleScript $ fromAllegraTimelock s
         Alonzo.PlutusScript Alonzo.PlutusV1 s ->
-          ScriptInEra PlutusScriptV1InAlonzo $
-          PlutusScript PlutusScriptV1 $
-          PlutusScriptSerialised s
+          ScriptInEra PlutusScriptV1InAlonzo
+            . PlutusScript PlutusScriptV1
+            $ PlutusScriptSerialised s
         Alonzo.PlutusScript Alonzo.PlutusV2 _ ->
           error "fromShelleyBasedScript: PlutusV2 not supported in Alonzo era"
     ShelleyBasedEraBabbage ->
       case script of
         Alonzo.TimelockScript s ->
-          ScriptInEra SimpleScriptV2InBabbage $
-          SimpleScript SimpleScriptV2 $
-          fromAllegraTimelock TimeLocksInSimpleScriptV2 s
+          ScriptInEra SimpleScriptInBabbage
+            . SimpleScript $ fromAllegraTimelock  s
         Alonzo.PlutusScript Alonzo.PlutusV1 s ->
-          ScriptInEra PlutusScriptV1InBabbage $
-          PlutusScript PlutusScriptV1 $
-          PlutusScriptSerialised s
+          ScriptInEra PlutusScriptV1InBabbage
+            . PlutusScript PlutusScriptV1 $ PlutusScriptSerialised s
         Alonzo.PlutusScript Alonzo.PlutusV2 s ->
-          ScriptInEra PlutusScriptV2InBabbage $
-          PlutusScript PlutusScriptV2 $
-          PlutusScriptSerialised s
+          ScriptInEra PlutusScriptV2InBabbage
+            . PlutusScript PlutusScriptV2 $ PlutusScriptSerialised s
 
 
 
 -- | Conversion for the 'Shelley.MultiSig' language used by the Shelley era.
 --
-toShelleyMultiSig :: SimpleScript SimpleScriptV1
-                  -> Shelley.MultiSig StandardCrypto
+toShelleyMultiSig :: SimpleScript -> Shelley.MultiSig StandardCrypto
 toShelleyMultiSig = go
   where
-    go :: SimpleScript SimpleScriptV1 -> Shelley.MultiSig StandardCrypto
+    go :: SimpleScript -> Shelley.MultiSig StandardCrypto
     go (RequireSignature (PaymentKeyHash kh))
                         = Shelley.RequireSignature (Shelley.coerceKeyRole kh)
     go (RequireAllOf s) = Shelley.RequireAllOf (map go s)
     go (RequireAnyOf s) = Shelley.RequireAnyOf (map go s)
     go (RequireMOf m s) = Shelley.RequireMOf m (map go s)
+    go timelock = error $ "toShelleyMultiSig: " <> show timelock <>
+                          " not supported in MultiSig scripts."
```

Comment:
I'm wondering if it might be better not to `error` here.

Using `error` from the CLI is okay, but `error` would cause an application that uses the API to exit.

Reply:
So I opted to return an `Either` as you suggested for `toShelleyMultiSig`. However `toShelleyMultiSig` is called in: `toShelleyScript :: ScriptInEra era -> Ledger.Script (ShelleyLedgerEra era)`. I opted to error here instead because this will `error` if someone is running a Shelley era network and tries to use a Timelock. I think the odds of this is extremely low and not worth polluting the code further with an `Either`. Lmk what you think.

Reply:
I think this is okay to merge, but I also that we should create a ticket to remove as many `error` calls as we reasonably can.

Reply:
I think let this pass for now given the low probably of it being a problem, but we should also create a ticket to remove calls to `error` from `cardano-api` as a separate piece of work to be done after the next release.